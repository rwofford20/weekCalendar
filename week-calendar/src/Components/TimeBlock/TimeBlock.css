.timeblock-container {
    /* outline: 1px dotted royalblue; */
    background-color: rgb(163, 219, 238);
    border-radius: var(--time-block-border-radius);
    margin: .1rem;
    color:rgb(6, 55, 95);
}

.timeblock-title-container, .timeblock-time-container {
    border-radius: var(--time-block-border-radius) var(--time-block-border-radius) 0  0;
    padding: .1rem .35rem;
    padding: .1rem .35rem;
    font-size: .9rem;
}

.timeblock-title-container {
    background-color: inherit;
    mix-blend-mode: multiply;
}