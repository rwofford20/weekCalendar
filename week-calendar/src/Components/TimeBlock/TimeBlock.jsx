import React from 'react';
import './TimeBlock.css';

const TimeBlock = (props) => (
    <div>
        {props.title}
        {props.startTime}
        {props.endTime}
    </div>
);

export default TimeBlock;