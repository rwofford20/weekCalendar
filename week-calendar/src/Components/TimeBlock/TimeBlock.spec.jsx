import React from 'react';
import {shallow } from 'enzyme';
import TimeBlock from './TimeBlock';

describe('TimeBlock', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<TimeBlock />));

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('it should render a div', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('it should render a title', () => {
        wrapper.setProps({ title: 'Sample'});
        expect(wrapper.text()).toEqual('Sample');
    });

    it('it should render a start time', () => {
        wrapper.setProps({ startTime: '900'});
        expect(wrapper.text()).toEqual('900');
    });

    it('it should render an end time', () => {
        wrapper.setProps({ endTime: '1000'});
        expect(wrapper.text()).toEqual('1000');
    });
});