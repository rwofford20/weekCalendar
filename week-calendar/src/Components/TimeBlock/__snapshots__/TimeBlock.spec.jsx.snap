// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeBlock should render correctly 1`] = `ShallowWrapper {}`;
